### DISCUSSION TITLE
Adding additional assemblies to index from binary data (COFF-based image)

### DISCUSSION THREAD

**Discussion by montyclt** [ravendb/ravendb#14190] [Q&A]
To load additional assemblies into index, there are three methods:
- `FromRuntime`
- `FromPath`
- `FromNuGet`

There are any form to load it from a byte array or stream? Should I need to save the file into temp directory and use `FromPath`?

**Comment by ayende** [comment]
Where do you expect to get that binary _from_ ?

**Reply by montyclt** [reply]
In my app, it comes from a stream that comes from zip file.

**Reply by ayende** [reply]
You want to add that during the creation of the index?
That isn't supported. You _can_ play some tricks (have an additional source that dynamically load from base 64.
Please don't do that. 
Easiest is to add that in a well known path.

**Reply by montyclt** [reply]
Yes, I want to upload assembly from my application to RavenDB server during index creation.

I think that I can copy the dll files to a shared volume during index creation...

**Reply by ayende** [reply]
That, or a private Nuget package are probably the best options.
I would strongly recommend that over other attempts like this.

**Reply by montyclt** [reply]
Thanks Ayende. Only for curiosity, why uploading assembly from client is not supported?

**Reply by ayende** [reply]
Super complex to actually do.

To start with, you are typically not interested in just one assembly. You need referenced assembly, resources, etc.
How do you handle this? 

Then you have to worry about all _sorts_ of fun stuff. What happens when you shut down the db? And then load it again (in the same process). What happens across the cluster?
How do you update the assembly?

Finally, there are also issues with the assembly context, which can cause _Weird_ errors.
